src/Verify/IoHelpers.cs

```diff
@@ -203,6 +203,24 @@ public static string GetRelativePath(string directory, string file)
 
     }
 
+    static VirtualizedRunHelper? virtualizedRunHelper;
+
+    internal static void MapPathsForCallingAssembly(Assembly assembly) =>
+        virtualizedRunHelper = new(assembly);
+
+    internal static string? GetMappedBuildPath(string? path, Assembly? assembly = null)
+    {
+        if (virtualizedRunHelper == null && assembly != null)
```

Comment:
this is problematic since based on people use of `AttributeReader` the `virtualizedRunHelper ` can be mutated. shouldnt the static instance of `virtualizedRunHelper` only be initialized in `TargetAssembly.Assign`?

Reply:
Correct - I originally intentionaly had it that way - as I was expecting all the callers to `AttributeReader` and `Verify` to be located in a code that is all going through single build in a single location. While the code can be in multiple assemblies, it still would have single root (solution) - so despite performing mapping via different projects/assemblies, result would be same.
Would the code be comming from different builds - then hopes for proper mapping of every single path are lost due to inability to locate and reach the other build code location from the virtualized mapped path.

...

At least that was my thinking.. but based on your comment I realized there is one specific scenario where separating mappers per assembly makes sense - and that's when the callers are in different assemblies, comming from different builds, but all those builds occured on the same developer machine and from same drive. The code roots might differ. All the code is still reachable from the VM/docker then and the mapping code should be able to reflect that.

A bit esoteric scenario (yeah - I'm the last one to speak about esoteric scenarios), but not nonexistent, plus code is cleaner this way. So thanks for detailed look!

**tl;dr;:** **I have updated the code to separate mappers per assembly.**